pkg/shared/util/string.go

```diff
@@ -48,3 +48,31 @@ func StringSliceContains(list []string, str string) bool {
 	}
 	return false
 }
+
+func CompareSlice(operator string, sa []string, sb []string) bool {
+	switch operator {
+	case "and":
+		for _, val := range sa {
+			if !StringSliceContains(sb, val) {
+				return false
+			}
+		}
+		return true
```

Comment:
If we construct a string hash set of slice b before `switch`, we can achieve O(n) time with O(n) space. Not sure if that could be a performance improvement. 😃

Reply:
@yhl25 Please consider my proposal of saving time complexity by adding space complexity (GoLang GC periodically reclaims the allocated memory).

`CompareSlice` is on a frequent path that every message goes through it hence its performance matters.

Reply:
+1, also, the tags are constant; we can build that hash at the initialization phase. 
Let's do that via a follow-up separate PR, as this has a lot of breaking changes.

Reply:
#672

Reply:
yeah @ashwinidulams and I had discussed using a hashmap for tags, but since the number of keys is very less we thought don't need it. But since compareSlice gets invoked for each message the optimisation might actually help, will fix it in a follow-up PR

Reply:
Giving it a second thought, it might not help much to introduce a hash map. O(n) matters when n is large but for `CompareSlice`, we expect both input slices to be small. We can dive deeper in #672 . Thank you Vigith for creating the issue. :)